### DISCUSSION TITLE
Client Certificate support

### DISCUSSION THREAD

**Discussion by darakian** [diesel-rs/diesel#2665] [Ideas]
Hey All,

Is it feasible to add client certificate support as an auth mechanism for at least Postgres. Looking through the docs it seems that only passwords are currently supported. Postgres has supported client certs for quite a while and it's a much nicer auth mechanism to orchestrate with (imo).

Postgres docs:
https://www.postgresql.org/docs/13/ssl-tcp.html#SSL-CLIENT-CERTIFICATES

**Comment by weiznich** [comment]
Its already possible to do that via libpq connection urls. See [their](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING) documentation for details.

**Reply by darakian** [reply]
Oh I see. Thanks for the link I missed that.
